rust/projects/tools/tests/tests.rs

```diff
@@ -38,3 +90,33 @@ fn get_non_existent_value() {
     store.set("key1".to_owned(), "value1".to_owned());
     assert_eq!(store.get("key2".to_owned()), None);
 }
+
+// Path to kvs binary
+fn binary_path() -> PathBuf {
+    // Path to cargo executables
+    // Adapted from https://github.com/rust-lang/cargo/blob/485670b3983b52289a2f353d589c57fae2f60f82/tests/testsuite/support/mod.rs#L507
+    let mut path = current_exe()
+        .ok()
+        .map(|mut path| {
+            path.pop();
+            if path.ends_with("deps") {
+                path.pop();
+            }
+            path
+        })
+        .unwrap();
+
+    path.push("kvs");
+    path
+}
```

Comment:
Clever to take this from cargo. This is a better solution than I would have written.